ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -584,4 +592,44 @@ public void setHitSlopRect(@Nullable Rect rect) {
     mHitSlopRect = rect;
   }
 
+  public void setOpacityIfPossible(float opacity) {
+    mBackfaceOpacity = opacity;
+    setBackfaceVisibilityDependantOpacity();
+  }
+
+  public void setBackfaceVisibility(String backfaceVisibility) {
+    mBackfaceVisibility = backfaceVisibility;
+    setBackfaceVisibilityDependantOpacity();
+  }
+
+  public void decomposeMatrix(ReadableArray matrix) {
+    if (matrix != null) {
+      TransformHelper.processTransform(matrix, sTransformDecompositionArray);
+      MatrixMathHelper.decomposeMatrix(sTransformDecompositionArray, sMatrixDecompositionContext);
+      setBackfaceVisibilityDependantOpacity();
+    }
+  }
+
+  private void setBackfaceVisibilityDependantOpacity() {
+    boolean isBackfaceVisible = mBackfaceVisibility.equals("visible");
+    boolean isRotationInformationAvailable = sMatrixDecompositionContext != null;
+
+    if (isBackfaceVisible || !isRotationInformationAvailable) {
+      setAlpha(mBackfaceOpacity);
+      return;
+    }
+
+    float rotationX = (float) sMatrixDecompositionContext.rotationDegrees[0];
+    float rotationY = (float) sMatrixDecompositionContext.rotationDegrees[1];
+
+    boolean isFrontfaceVisible = (rotationX >= -90.f && rotationX < 90.f) &&
```

Comment:
Could a negative scale affect the backface visibility?

Reply:
Negative degrees are handled.

```javascript
flipAnimation.interpolate({
  inputRange: [0, 180],
  outputRange: ['0deg', '180deg'],
});
```

```javascript
flipAnimation.interpolate({
  inputRange: [0, 180],
  outputRange: ['0deg', '-180deg'],
});
```